<launch>

 

    <!-- Laser scan matcher -->
  <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" 
    name="Laser_scan_matcher" output="screen">

    <!-- Topics -->
    <remap from="scan" to="front_laser/scan"/>
    <!-- <remap from="cloud" to="/camera/depth/points"/> -->
    <!-- <remap from="imu/data" to="imu/data_compass"/> -->
    <remap from="odom" to="/mechROS/odom"/>
    <!-- <remap from="vel" to="/mechROS/base_control/cmd_vel" /> -->
    <remap from="pose_with_covariance_stamped" to="/Laser_scan_matcher/pose"/>

    <!-- Coordinate frames -->
    <param name="fixed_frame" value="/mechROS/tf/odom" />
    <param name="base_frame" value="/mechROS_base_link" />

    <!-- Motion prediction -->
    <param name="use_imu" value="true"/>
    <param name="use_odom" value="false"/>
    <param name="use_vel" value="false"/>

    <!-- Point cloud input -->
    <param name="use_cloud_input" value="false"/>
    <!-- <param name="cloud_range_min" value="0.4"/> -->
    <!-- <param name="cloud_range_max" value="10.0"/> -->

    <!-- Key frames -->
    <param name="kf_dist_linear" value="0.07"/>
    <param name="kf_dist_angular" value="0.15"/>

    <!-- Output -->
    <param name="publish_tf" value="false"/>
    <param name="publish_pose" value="false"/>
    <param name="publish_pose_with_covariance" value="false"/>
    <param name="publish_pose_with_covariance_stamped" value="true"/>

    <!-- Scan matching -->
    <param name="max_iterations" value="50"/>
    <param name="max_correspondence_dist" value="0.3"/>
    <param name="max_angular_correction_deg" value="45.0"/>
    <param name="max_linear_correction" value="0.5"/>
    <param name="epsilon_xy" value="0.000001"/>
    <param name="epsilon_theta" value="0.000001"/>
    <param name="outliers_maxPerc" value="0.90"/>

    <!-- Scan matching advanced -->
    <param name="do_compute_covariance" value="0"/>
    <rosparam param="position_covariance">[ 0.005,  0.005,  0.005]</rosparam>
    <rosparam param="orientation_covariance">[ 0.005,  0.005,  0.005]</rosparam>


  </node>

  
</launch>